lib/rucio/tests/test_meta.py

```diff
@@ -83,3 +88,63 @@ def xtest_add_value_to_bad_key(self):
     def xtest_add_key_mysql(self):
         """ META (CORE): Add a new key to test conversions on MySQL"""
         self.meta_client.add_key('datatype', DIDType.FILE)
+
+    def xtest_insert_metadata(self):
+        """ META (CLIENTS) : Adds a fully set json column to a did (rewrites if already present) """
+        data = {"key1": "value_" + str(uuid()), "key2": "value_" + str(uuid()), "key3": "value_" + str(uuid())}
+        self.meta_client.insert_metadata(scope=self.tmp_scope, name=self.tmp_name, meta=data)
+
+    def xtest_update_metadata_key(self):
+        """ META (CLIENTS) : Updates metadata key, if not present creates a new one """
+        data = {"key1": "value_" + str(uuid()), "key2": "value_" + str(uuid()), "key3": "value_" + str(uuid())}
+        self.meta_client.insert_metadata(scope=self.tmp_scope, name=self.tmp_name, meta=data)
+
+        key = "key1"
+        value = 'value_' + str(uuid())
+        self.meta_client.update_key(scope=self.tmp_scope, name=self.tmp_name, key=key, value=value)
+
+    def xtest_update_metadata_keys(self):
+        """ META (CLIENTS) : Updates multiple metadata keys, if not present creates new ones """
+        data = {"key1": "value_" + str(uuid()), "key2": "value_" + str(uuid()), "key3": "value_" + str(uuid())}
+        self.meta_client.insert_metadata(scope=self.tmp_scope, name=self.tmp_name, meta=data)
+
+        data = {"key4": "value_" + str(uuid()), "key5": "value_" + str(uuid())}
+        self.meta_client.update_keys(scope=self.tmp_scope, name=self.tmp_name, data=data)
```

Comment:
isn't  `update_metadata` a better name? also, is it idempotent with add_metadata? if so, it should be collapsed into a single function.